### PR TITLE
FWI-5804 - expose issue fixer and mutations in the library

### PR DESCRIPTION
**Pull request by vitorvezani**
This PR fixes #
- Expose `fix.Execute`
- Fix migration generation even if the check passes
- [minor] Fix error handling and typo

## Checklist
* [X] I have signed the CLA
* [X] I have updated/added any relevant documentation

### CODE REVIEW

File: pkg/kube/resources.go [hunk 1]
```diff
@@ -215,10 +215,13 @@ func CreateResourceProviderFromPath(directory string) (*ResourceProvider, error)
 }
 
 // CreateResourceProviderFromYaml returns a new ResourceProvider using the yaml
-func CreateResourceProviderFromYaml(yamlContent string) *ResourceProvider {
+func CreateResourceProviderFromYaml(yamlContent string) (*ResourceProvider, error) {
 	resources := newResourceProvider("unknown", "Content", "unknown")
-	resources.addResourcesFromYaml(string(yamlContent))
-	return &resources
+	err := resources.addResourcesFromYaml(string(yamlContent))
```
Comment (vitorvezani) [inline pkg/kube/resources.go:220]: it was ignoring returned `err`

File: pkg/validator/schema.go [hunk 2]
```diff
@@ -35,7 +35,7 @@ import (
 type schemaTestCase struct {
 	Target           config.TargetKind
 	Resource         kube.GenericResource
-	IsInitContianer  bool
+	IsInitContainer  bool
```
Comment (vitorvezani) [inline pkg/validator/schema.go:38]: minor typo

File: pkg/validator/schema.go [hunk 3]
```diff
@@ -408,15 +408,13 @@ func applySchemaCheck(conf *config.Configuration, checkID string, test schemaTes
 
 	}
 	result := makeResult(conf, check, passes, issues)
-	if !passes {
```
Comment (vitorvezani) [inline pkg/validator/schema.go:411]: We do want to know the retrieve the `mutations`, even if the check passes

File: cmd/polaris/fix.go [hunk 4]
```diff
@@ -52,145 +43,16 @@ var fixCommand = &cobra.Command{
...
 		logrus.Debug("Setting up controller manager")
 
-		if filesPath == "" {
-			logrus.Error("Please specify a files-path flag")
-			cmd.Help()
-			os.Exit(1)
-		}
-		var yamlFiles []string
-		fileInfo, err := os.Stat(filesPath)
+		err := fix.Execute(config, filesPath, isTemplate, checksToFix...)
```
Comment (vitorvezani) [inline cmd/polaris/fix.go:46]: Expose `fix.Execute` for being used as lib.

Comment (rbren) [reply cmd/polaris/fix.go:46]: There are no breaking API changes to the `fix` command, correct?

**Review by rbren — APPROVED** [review]

File: cmd/polaris/fix.go (hunk 4 above)
Comment (vitorvezani) [reply cmd/polaris/fix.go:46]: correct

**Pull request merged**